sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/AesKw.cs

```diff
@@ -0,0 +1,422 @@
+            {
+                get { throw new NotImplementedException(); }
+            }
+
+            public int OutputBlockSize
+            {
+                get { throw new NotImplementedException(); }
+            }
+
+            public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
+            {
+                // No support for block-by-block transformation
+                throw new NotImplementedException();
+            }
+
+            public byte[] TransformFinalBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+            {
+                if (inputBuffer == null)
+                    throw new ArgumentNullException(nameof(inputBuffer));
+
+                if (inputBuffer.Length == 0)
+                    throw new ArgumentNullException(nameof(inputBuffer), "The length of the inputBuffer parameter cannot be zero");
+
+                if (inputCount <= 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The inputCount parameter must not be zero or negative");
+
+                if (inputCount % 8 != 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The inputCount parameter must be a multiple of 64 bits");
+
+                if (inputOffset < 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputOffset), "The inputOffset parameter must not be negative");
+
+                if (inputOffset + inputCount > inputBuffer.Length)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The sum of inputCount and inputOffset parameters must not be larger than the length of inputBuffer");
+
+                /*
+                   1) Initialize variables.
+
+                       Set A = IV, an initial value (see 2.2.3)
+                       For i = 1 to n
+                           R[i] = P[i]
+
+                   2) Calculate intermediate values.
+
+                       For j = 0 to 5
+                           For i=1 to n
+                               B = AES(K, A | R[i])
+                               A = MSB(64, B) ^ t where t = (n*j)+i
+                               R[i] = LSB(64, B)
+
+                   3) Output the results.
+
+                       Set C[0] = A
+                       For i = 1 to n
+                           C[i] = R[i]
+                */
+
+                // The default initialization vector from RFC3394
+                byte[] a = _iv;
+
+                // The number of input blocks
+                var n = inputCount >> 3;
+
+                // The set of input blocks
+                byte[] r = new byte[n << 3];
+
+                Array.Copy(inputBuffer, inputOffset, r, 0, inputCount);
+
+                var encryptor = _aes.CreateEncryptor();
+                byte[] block = new byte[16];
+
+                // Calculate intermediate values
+                for (var j = 0; j < 6; j++)
```

Comment:
Nit: just to better match up with the documented algorithm, should this maybe be "s" or "steps"?

Reply:
This code is ported from directly track 1 and I'm leaving all the code in this class as is, since it's already been security reviewed for it's crypto implementation.